backend/dal/Services/SecurityDepositReturnService.cs

```diff
@@ -0,0 +1,68 @@
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Services
+{
+    public class SecurityDepositReturnService : ISecurityDepositReturnService
+    {
+        readonly ISecurityDepositReturnRepository _securityDepositReturnRepository;
+        readonly ILeaseRepository _leaseRepository;
+        readonly ILeaseService _leaseService;
+        readonly ClaimsPrincipal _user;
+        public SecurityDepositReturnService(ISecurityDepositReturnRepository depositReturnRepository, ILeaseRepository leaseRepository, ILeaseService leaseService, ClaimsPrincipal user)
+        {
+            _securityDepositReturnRepository = depositReturnRepository;
+            _leaseRepository = leaseRepository;
+            _leaseService = leaseService;
+            _user = user;
+        }
+
+        public PimsLease AddLeaseDepositReturn(long leaseId, long leaseRowVersion, PimsSecurityDepositReturn deposit)
+        {
+            ValidateServiceCall(leaseId, leaseRowVersion);
+            deposit.LeaseId = leaseId;
```

Comment:
is this necessary? upstream of this in the controller you throw an exception if the lease id is mismatched right? maybe that logic should be here instead?

Reply:
fair enough, updated in the other places as well